createtransfers/createtransfers.py

```diff
@@ -194,17 +199,137 @@ def create_variously_encoded_files():
+    tool.
+    """
+    test_transfers = "TestTransfers"
+    sample_packages = "sample-zip-packages"
+
+    # Control the name of these top-level folders so tat we know that we can
+    # recurse into them to zip the various contents.
+    var_encoded_files = "variously-encoded-files"
+    var_encoded_dirs = "variously-encoded-dirs"
+    var_deep_transfer = "deep-transfer"
+    var_deep_zip_transfer = os.path.join(var_deep_transfer,
+                                         "deep_zip_transfer")
+    var_large_transfer = "large-transfer"
+    var_large_zip_transfer = os.path.join(var_large_transfer,
+                                          "large_zip_transfer")
+
+    # Create zips with different file name encodings in each.
+    zip_path_files = os.path.join(HERE, test_transfers, sample_packages,
+                                  var_encoded_files)
+    create_variously_encoded_files(zip_path_files)
+    create_zip_dance(zip_path_files)
+
+    # Create zips with different directory name encodings in each.
+    zip_path_dirs = os.path.join(HERE, test_transfers, sample_packages,
+                                 var_encoded_dirs)
+    create_variously_encoded_dir_names(zip_path_dirs)
+    create_zip_dance(zip_path_dirs)
+
+    # Create a single zip with one deep directory structure inside using the
+    # application defaults.
+    zip_path_deep = os.path.join(HERE, test_transfers, sample_packages,
+                                 var_deep_zip_transfer)
+    create_deep_transfers(
+        zip_path=zip_path_deep,
+        recursion_depth=createtransfersargs.DEFAULT_DEPTH,
+        number_of_directories=createtransfersargs.DEFAULT_NUMBER_DIRS,
+        number_of_files=createtransfersargs.DEFAULT_NUMBER_FILES)
+    # The first directory above is deliberately made one additional directory
+    # down so that it can be zipped as one. First we have to come back up a
+    # level which we do here.
+    one_deep_zip_path = os.path.join(HERE, test_transfers, sample_packages,
+                                     var_deep_transfer)
+    create_zip_dance(one_deep_zip_path)
+
+    # Create a path where we will create a large zipped transfer folder.
+    zip_path_large = os.path.join(HERE, test_transfers, sample_packages,
+                                  var_large_zip_transfer)
+    # Override the defaults to create a modest 1.1GB zip.
+    create_large_test_transfers(zip_path=zip_path_large,
+                                image_n=35,
+                                video_n=35)
+    # Like the deep transfer, we deliberately drop the large transfers down a
+    # folder level. We come back a level here to zip it all at once.
+    one_large_zip_path = os.path.join(HERE, test_transfers, sample_packages,
+                                      var_large_transfer)
+    create_zip_dance(one_large_zip_path)
+
+
+def create_zip_dance(base_directory):
+    """Perform various manouvers to create zip files from our content and move
```

Comment:
typo: "manouvers" should be "manoeuvers".

Reply:
:+1: